### PR TITLE
fix: removed wildcard pattern type in documentation

### PR DESCRIPTION
**Pull request by emasab**
Removed non-existent wildcard pattern type in documentation and explained it better.

### CODE REVIEW

File: kafka/adminapi.go [hunk 1]
```diff
@@ -1416,9 +1416,10 @@ func (a *AdminClient) CreateACLs(ctx context.Context, aclBindings ACLBindings, o
 //  * `aclBindingFilter` - A filter with attributes that must match.
 //     string attributes match exact values or any string if set to empty string.
 //     Enum attributes match exact values or any value if ending with `Any`.
-//     If `ResourcePatternType` is set to `ResourcePatternTypeMatch` returns all
-//     the ACL bindings with `ResourcePatternTypeLiteral`, `ResourcePatternTypeWildcard`
-//     or `ResourcePatternTypePrefixed` pattern type that match the resource name.
+//     If `ResourcePatternType` is set to `ResourcePatternTypeMatch` returns ACL bindings with:
+//     - `ResourcePatternTypeLiteral` pattern type with resource name equal to the given resource name
+//     - `ResourcePatternTypeLiteral` pattern type with wildcard resource name that matches the given resource name
```
Comment (mhowlett) [inline kafka/adminapi.go:1421]: do you mean ResourcePatternTypeWildcard here?

File: kafka/adminapi.go [hunk 2]
```diff
@@ -1471,9 +1472,10 @@ func (a *AdminClient) DescribeACLs(ctx context.Context, aclBindingFilter ACLBind
 //  * `aclBindingFilters` - a slice of ACL binding filters to match ACLs to delete.
 //     string attributes match exact values or any string if set to empty string.
 //     Enum attributes match exact values or any value if ending with `Any`.
-//     If `ResourcePatternType` is set to `ResourcePatternTypeMatch` returns all
-//     the ACL bindings with `ResourcePatternTypeLiteral`, `ResourcePatternTypeWildcard`
-//     or `ResourcePatternTypePrefixed` pattern type that match the resource name.
+//     If `ResourcePatternType` is set to `ResourcePatternTypeMatch` deletes ACL bindings with:
+//     - `ResourcePatternTypeLiteral` pattern type with resource name equal to the given resource name
+//     - `ResourcePatternTypeLiteral` pattern type with wildcard resource name that matches the given resource name
```
Comment (mhowlett) [inline kafka/adminapi.go:1477]: same as above

**Review by mhowlett — APPROVED** [review]
with suggested change considered.

File: kafka/adminapi.go (hunk 1 above)
Comment (emasab) [reply kafka/adminapi.go:1421]: No, in reality ResourcePatternTypeWildcard as a enum value doesn't exist, I introduced it in the documentation for an error when changing it from plain english to references to the enum values, if you look at the java [constants](https://kafka.apache.org/32/javadoc/org/apache/kafka/common/resource/PatternType.html) it's like that. The java documentation is [here](https://kafka.apache.org/32/javadoc/org/apache/kafka/common/resource/PatternType.html#MATCH)

**Pull request merged**